operation/fhir-operation-everything/src/test/java/com/ibm/fhir/operation/everything/EverythingOperationTest.java

```diff
@@ -0,0 +1,142 @@
+     * @throws FHIRParserException 
+     */
+    @Test
+    public void testConvertParametersStartEnd() throws FHIRParserException, IOException {
+        Parameters parameters = loadParametersFile("parameters-start-end.json");
+        MultivaluedMap<String, String> queryParameters = exportOperation.parseQueryParameters(parameters);
+        assertEquals(SearchConstants.MAX_PAGE_SIZE + "", queryParameters.getFirst(SearchConstants.COUNT));
+        assertEquals(2, queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).size());
+        assertTrue(queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).contains(EverythingOperation.STARTING_FROM + "1970-11-04"));
+        assertTrue(queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).contains(EverythingOperation.UP_UNTIL+ "1971-01-01"));
+    }
+
+    /**
+     * @throws IOException 
+     * @throws FHIRParserException 
+     */
+    @Test
+    public void testConvertParametersStartEndCount() throws FHIRParserException, IOException {
+        Parameters parameters = loadParametersFile("parameters-start-end-count.json");
+        MultivaluedMap<String, String> queryParameters = exportOperation.parseQueryParameters(parameters);
+        assertEquals("10", queryParameters.getFirst(SearchConstants.COUNT));
+        assertEquals(2, queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).size());
+        assertTrue(queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).contains(EverythingOperation.STARTING_FROM + "1970-11-04"));
+        assertTrue(queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).contains(EverythingOperation.UP_UNTIL+ "1971-01-01"));
+    }
+
+    /**
+     * @throws IOException 
+     * @throws FHIRParserException 
+     */
+    @Test
+    public void testConvertParametersStartEndCountSince() throws FHIRParserException, IOException {
+        Parameters parameters = loadParametersFile("parameters-start-end-count-since.json");
+        MultivaluedMap<String, String> queryParameters = exportOperation.parseQueryParameters(parameters);
+        assertEquals("10", queryParameters.getFirst(SearchConstants.COUNT));
+        assertEquals(2, queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).size());
+        assertTrue(queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).contains(EverythingOperation.STARTING_FROM + "1970-11-04"));
+        assertTrue(queryParameters.get(EverythingOperation.DATE_QUERY_PARAMETER).contains(EverythingOperation.UP_UNTIL+ "1971-01-01"));
+        assertEquals(EverythingOperation.STARTING_FROM + "2017-01-01T00:00Z", queryParameters.getFirst(EverythingOperation.LAST_UPDATED_QUERY_PARAMETER));
+    }
+
+    private Parameters loadParametersFile(String file) throws IOException, FHIRParserException {
+        try (InputStreamReader reader = new InputStreamReader(this.getClass().getResourceAsStream("/" + file))) {
+            return FHIRParser.parser(Format.JSON).parse(reader);
+        }
+    }
+}
```

Comment:
Maybe add some error condition tests

Reply:
If I test passing in bad input here, I am not really testing my code which wouldn't even be reached, I would be testing the code listed here which I assume has already been unit tested
![image](https://user-images.githubusercontent.com/1807320/110180238-a464d400-7dc6-11eb-9bcd-b528f49a87b0.png)